src/inference/graph_inference.cpp

```diff
@@ -88,6 +88,14 @@ namespace std {
   };
 }
 
+uint64 HashInt(int x) {
```

Comment:
This is definitely buggy: x is int, you calculate all the time 32 bit and then use it as 64 bit.